File: backend/job-service/src/main/java/com/recruitpme/jobservice/controller/QuestionController.java
package com.recruitpme.jobservice.controller;

import com.recruitpme.jobservice.dto.QuestionDTO;
import com.recruitpme.jobservice.dto.QuestionSetDTO;
import com.recruitpme.jobservice.service.QuestionService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;
import java.util.List;

@RestController
@RequestMapping("/api/questions")
@RequiredArgsConstructor
public class QuestionController {

    private final QuestionService questionService;

    @GetMapping("/custom")
    public ResponseEntity<List<QuestionDTO>> getAllQuestions() {
        return ResponseEntity.ok(questionService.getAllQuestions());
    }

    @GetMapping("/{id}")
    public ResponseEntity<QuestionDTO> getQuestionById(@PathVariable String id) {
        return ResponseEntity.ok(questionService.getQuestionById(id));
    }

    @PostMapping
    public ResponseEntity<QuestionDTO> createQuestion(@Valid @RequestBody QuestionDTO questionDTO) {
        return ResponseEntity.status(HttpStatus.CREATED).body(questionService.createQuestion(questionDTO));
    }

    @PutMapping("/{id}")
    public ResponseEntity<QuestionDTO> updateQuestion(
            @PathVariable String id,
            @Valid @RequestBody QuestionDTO questionDTO) {
        return ResponseEntity.ok(questionService.updateQuestion(id, questionDTO));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteQuestion(@PathVariable String id) {
        questionService.deleteQuestion(id);
        return ResponseEntity.noContent().build();
    }

    @GetMapping("/sets")
    public ResponseEntity<List<QuestionSetDTO>> getAllQuestionSets() {
        return ResponseEntity.ok(questionService.getAllQuestionSets());
    }

    @GetMapping("/sets/{id}")
    public ResponseEntity<QuestionSetDTO> getQuestionSetById(@PathVariable String id) {
        return ResponseEntity.ok(questionService.getQuestionSetById(id));
    }

    @PostMapping("/sets")
    public ResponseEntity<QuestionSetDTO> createQuestionSet(@Valid @RequestBody QuestionSetDTO questionSetDTO) {
        return ResponseEntity.status(HttpStatus.CREATED).body(questionService.createQuestionSet(questionSetDTO));
    }

    @PutMapping("/sets/{id}")
    public ResponseEntity<QuestionSetDTO> updateQuestionSet(
            @PathVariable String id,
            @Valid @RequestBody QuestionSetDTO questionSetDTO) {
        return ResponseEntity.ok(questionService.updateQuestionSet(id, questionSetDTO));
    }

    @DeleteMapping("/sets/{id}")
    public ResponseEntity<Void> deleteQuestionSet(@PathVariable String id) {
        questionService.deleteQuestionSet(id);
        return ResponseEntity.noContent().build();
    }
}

File: backend/src/api/questions/question.controller.js
import QuestionService from './question.service.js';

class QuestionController {
  async getCustomQuestions(req, res, next) {
    try {
      const { companyId } = req.params;
      const questions = await QuestionService.getCustomQuestions(req.user.id, companyId);
      res.status(200).json(questions);
    } catch (error) {
      next(error);
    }
  }

  async getQuestionSets(req, res, next) {
    try {
      const { companyId } = req.params;
      const sets = await QuestionService.getQuestionSets(req.user.id, companyId);
      res.status(200).json(sets);
    } catch (error) {
      next(error);
    }
  }

  async createQuestion(req, res, next) {
    try {
      const { companyId } = req.params;
      const question = await QuestionService.createQuestion(req.user.id, companyId, req.body);
      res.status(201).json({ message: 'Question created.', data: question });
    } catch (error) {
      next(error);
    }
  }

  async updateQuestion(req, res, next) {
    try {
      const { companyId, id } = req.params;
      const updatedQuestion = await QuestionService.updateQuestion(req.user.id, companyId, id, req.body);
      res.status(200).json({ message: 'Question updated.', data: updatedQuestion });
    } catch (error) {
      next(error);
    }
  }

  async deleteQuestion(req, res, next) {
    try {
      const { companyId, id } = req.params;
      const result = await QuestionService.deleteQuestion(req.user.id, companyId, id);
      res.status(200).json(result);
    } catch (error) {
      next(error);
    }
  }

  async createQuestionSet(req, res, next) {
    try {
      const { companyId } = req.params;
      const set = await QuestionService.createQuestionSet(req.user.id, companyId, req.body);
      res.status(201).json({ message: 'Question set created.', data: set });
    } catch (error) {
      next(error);
    }
  }

  async updateQuestionSet(req, res, next) {
    try {
      const { companyId, id } = req.params;
      const updatedSet = await QuestionService.updateQuestionSet(req.user.id, companyId, id, req.body);
      res.status(200).json({ message: 'Question set updated.', data: updatedSet });
    } catch (error) {
      next(error);
    }
  }

  async deleteQuestionSet(req, res, next) {
    try {
      const { companyId, id } = req.params;
      const result = await QuestionService.deleteQuestionSet(req.user.id, companyId, id);
      res.status(200).json(result);
    } catch (error) {
      next(error);
    }
  }
}

export default new QuestionController();